libraries/chain/wasm_interface.cpp

```diff
@@ -1491,6 +1491,12 @@ class transaction_api : public context_aware_api {
       }
 
       void send_deferred( const uint128_t& sender_id, account_name payer, array_ptr<char> data, uint32_t data_len, uint32_t replace_existing) {
+         bool stop_deferred_transactions_activated = context.control.is_builtin_activated(builtin_protocol_feature_t::stop_deferred_transactions);
+         
+         if ( stop_deferred_transactions_activated && replace_existing ) {
```

Comment:
It's a bit odd that the handling for stop_deferred_transactions is split between here and schedule_deferred_transaction.

Reply:
This is consistent with protocol feature `builtin_protocol_feature_t::disallow_empty_producer_schedule`:
https://github.com/EOSIO/eos/blob/9cb5a2101a08d8c2d3e474409066bdfd42221a86/libraries/chain/wasm_interface.cpp#L205
https://github.com/EOSIO/eos/blob/9cb5a2101a08d8c2d3e474409066bdfd42221a86/libraries/chain/controller.cpp#L2947

Reply:
It's not really the same.  The way you've implemented it requires checks in both apply_context and wasm_interface to be correct.  My problem is that apply_context::schedule_deferred_transaction doesn't seem to have a clear understanding of its own preconditions.

Reply:
I think I need more clarification on this if it's still a problem.

Reply:
In retrospect, my complaint here was really just a restatement of the other issues I raised regarding send_deferred and schedule_deferred_transaction from a different perspective.